video/out/gpu/video.c

```diff
@@ -1291,8 +1299,22 @@ static void copy_image(struct gl_video *p, int *offset, struct image img)
         img.multiplier *= 1.0 / (tex_max - 1);
     }
 
-    GLSLF("color.%s = %f * vec4(texture(texture%d, texcoord%d)).%s;\n",
-          dst, img.multiplier, id, id, src);
+    if (img.interlaced) {
+        // Simple "blend" deinterlacer based on vlc's deinterlace/algo_basic.c
+        // The two fields are averaged together to produce one frame.
+        // Each line output is the input line averaged with the previous line.
+        // The very first line is copied exactly (albeit by averaging itself).
+        GLSLF("float yprev%d = texcoord%d.y == 0.0 ? 0.0 : "
+                              "texcoord%d.y - 1.0 / texture_size%d.y;\n",
```

Comment:
This clamp is unnecessary. Sampling out of bounds is fine. Also, `1.0 / texture_size%d.y` should be `texture_pt%d.y`

Reply:
(Also, in the future, use functions like `max()` or `clamp()` when clamping, rather than conditionals.)